File: typescript/src/odd-int/odd-int.ts
export function oddInt(arr: number[]): number {
  const counts = new Map<number, number>();

  for (const num of arr) {
    counts.set(num, (counts.get(num) || 0) + 1);
  }

  for (const [num, count] of counts.entries()) {
    if (count % 2 !== 0) {
      return num;
    }
  }

  return 0;
}

File: golang/odd-int/odd_int.go
package odd_int

func OddInt(arr []int) int {
	counts := make(map[int]int)
	for _, num := range arr {
		counts[num]++
	}

	for num, count := range counts {
		if count%2 != 0 {
			return num
		}
	}

	return 0
}
